lxd-agent/events.go

```diff
@@ -141,15 +140,78 @@ func eventsProcess(event api.Event) {
 		return
 	}
 
-	// Attempt to perform the mount.
-	mntSource := fmt.Sprintf("lxd_%s", e.Name)
+	if e.Action == "added" {
+		// Attempt to perform the mount.
+		mntSource := fmt.Sprintf("lxd_%s", e.Name)
 
-	_ = os.MkdirAll(e.Config["path"], 0755)
-	_, err = shared.RunCommand("mount", "-t", "virtiofs", mntSource, e.Config["path"])
-	if err != nil {
-		logger.Infof("Failed to mount hotplug %q (Type: %q) to %q", mntSource, "virtiofs", e.Config["path"])
+		_ = os.MkdirAll(e.Config["path"], 0755)
+		_, err = shared.RunCommand("mount", "-t", "virtiofs", mntSource, e.Config["path"])
+		if err != nil {
+			logger.Infof("Failed to mount hotplug %q (Type: %q) to %q", mntSource, "virtiofs", e.Config["path"])
+			return
+		}
+
+		logger.Infof("Mounted hotplug %q (Type: %q) to %q", mntSource, "virtiofs", e.Config["path"])
+	} else if e.Action == "removed" {
+		targetPath := e.Config["path"]
+
+		if !filepath.IsAbs(targetPath) {
+			logger.Infof("the device path must be absolute: %q", e.Config["path"])
+			return
+		}
+
+		file, err := os.Open("/proc/self/mountinfo")
+		if err != nil {
+			logger.Infof("Error opening /proc/self/mountinfo: %v\n", err)
+			return
+		}
+
+		defer file.Close()
+
+		var mountPoints []string
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			fields := strings.Fields(scanner.Text())
+			if len(fields) >= 10 && fields[4] == targetPath {
+				mountPoints = append(mountPoints, fields[4])
+			}
+		}
+
+		err = scanner.Err()
+		if err != nil {
+			logger.Infof("Error reading /proc/self/mountinfo: %v\n", err)
+			return
+		}
+
+		if len(mountPoints) == 0 {
+			logger.Infof("No mount points found for %s", targetPath)
+			return
+		}
+
+		if len(mountPoints) == 1 {
```

Comment:
We shouldn't need this special case, if there is only a single entry the general loop below should still be able to work to deal with it.

Reply:
fixed